pdl_syntax.md

```diff
@@ -0,0 +1,843 @@
+
+record UnionWithInlineRecord {
+  value = union[
+
+    record InlineRecord {
+      value: optional int
+    },
+
+    record InlineRecord2 {}
+  ]
+}
+```
+
+### Doc Strings
+
+Types and fields may be documented using “doc strings”.
+
+For example: 
+```
+/**
+ * Doc strings may be added to types. This doc should describe the purposes
+ * of the Example type.
+ */
+record Example {
+  /**
+   * Doc strings may also be added to fields.
+   */
+   field1: string
+
+   /** Doc strings can be single line.*/
+   field2: int
+}
+```
+
+**Note:**
+If you use Java comment style for doc string, e.g "// Doc String", those doc strings will not be stored in in-memory schema.
+
+### Including fields
+
+Record can include fields from one or more other records..
+
+For example: 
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * Bar includes fields of Foo, Bar will have fields b1 from itself and f1 from Foo
+ */
+record Bar includes Foo {
+  b1: string
+}
+```
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record Foo {
+  f1: string
+}
+```
+
+Multiple includes example:
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * Bar includes fields of Foo and Simple, Bar will have fields b1 from itself, f1 from Foo and s1 from Simple
+ */
+record Bar includes Foo, Simple {
+  b1: string
+}
+```
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record Simple {
+  s1: string
+}
+```
+
+In pegasus, field inclusion does not imply inheritance, it is merely a convenience to reduce duplication when writing schemas.
+
+
+### Properties
+Properties can be used to present arbitrary data and added to records, fields and enums.
+
+For example: 
+```
+@prop = "value"
+record Fruits {
+  @validate.regex.regex = "^(yes|no)$"
+  field: string
+}
+```
+#### Property values can be any valid JSON type:
+
+For example: 
+```
+@prop = 1
+```
+```
+@prop = "string"
+```
+```
+@prop = [1, 2, 3]
+```
+```
+@prop = { "a": 1", "b": { "c": true }}
+
+```
+#### Property values can also be empty: 
+
+If you don't indicate an explicity property value, it will resuilt in an implicity value of `true`.
+
+For example: 
+```
+namespace com.linkedin.pegasus.generator.examples
+
+@prop
+record Foo {
+  f1: string
+}
+```
+#### Property keys can be expressed as JSON:
+
+For example: 
+```
+@a = {
+  "b": {
+    "c": {
+      "d": {
+        "e": {
+          "f": false
+        }
+      }
+    }
+  }
+}
+```
+
+#### Property keys can be expressed as paths:
+
+The JSON style property key is complicated to write and read, so we provide a shorthand - the dot separate format to express the property keys.
+
+The following example is equivalent to the previous JSON example:
+
+```
+@a.b.c.d.e.f = false
+```
+
+### Escaping
+There are some keywords which are reserved in Pegasus. If you have to use them to define any names, you need to put them in backticks : ` `.
+
+#### Keyword Escaping
+
+```
+namespace com.linkedin.pegasus.generator.test.idl.escaping
```

Comment:
I can see that you copied many of these examples from the unit tests. The namespaces are a little lengthy. Wherever the namespace doesn't matter, make it something simple like `com.example.models`

Reply:
Updated